pandas/tests/groupby/test_transform.py

```diff
@@ -1074,3 +1072,57 @@ def test_transform_lambda_with_datetimetz():
         name="time",
     )
     assert_series_equal(result, expected)
+
+
+def test_transform_cumcount_ngroup():
+    df = DataFrame(dict(a=[0, 0, 0, 1, 1, 1], b=range(6)))
+    g = df.groupby(np.repeat([0, 1], 3))
+
+    # GH 27472
+    result = g.transform("cumcount")
+    expected = g.cumcount()
+    assert_series_equal(result, expected)
+
+    # GH 27468
+    result = g.transform("ngroup")
+    expected = g.ngroup()
+    assert_series_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    "func",
+    [
+        "backfill",
+        "bfill",
+        "cumcount",
+        "cummax",
+        "cummin",
+        "cumprod",
+        "cumsum",
+        "diff",
+        "ffill",
+        "pad",
+        "pct_change",
+        "rank",
+        "shift",
+        "ngroup",
+        pytest.param(
+            "fillna", marks=pytest.mark.xfail(reason="GH27905: potential bug")
+        ),
+        pytest.param(
+            "tshift", marks=pytest.mark.xfail(reason="GH27905: Should apply to ts data")
+        ),
+        pytest.param(
+            "corrwith",
+            marks=pytest.mark.xfail(reason="GH27905: Inapplicable to the data"),
+        ),
+    ],
+)
+def test_transformation_kernels_length(func):
+    # This test is to evaluate if after transformation, the index
+    # of transformed data is still the same with original DataFrame
+    df = DataFrame(dict(a=[0, 0, 0, 1, 1, 1], b=range(6)))
+    g = df.groupby(np.repeat([0, 1], 3))
+
+    result = g.transform(func)
+    assert (result.index == df.index).all()
```

Comment:
do we expect the same index object or just equality?

Reply:
yeah, should be more strict, changed to `assert_index_equal` to make sure index are same